interpreter/operator_arithmetic.go

```diff
@@ -960,6 +1042,411 @@ func maxValue(t types.IType, evaluationTimestamp *time.Time) (result.Value, erro
 // to UTC for min/max values, we use the evaluation timestamp's timezone.
 // We do this because when creating a literal it's also in the evaluation timestamp's timezone, and
 // some external tests will fail when these are different.
+// HighBoundary(input Decimal, precision Integer) Decimal
+// https://cql.hl7.org/09-b-cqlreference.html#highboundary
+func evalHighBoundaryDecimal(_ model.IBinaryExpression, input, precision result.Value) (result.Value, error) {
+	if result.IsNull(input) || result.IsNull(precision) {
+		return result.New(nil)
+	}
+	decValue, err := result.ToFloat64(input)
+	if err != nil {
+		return result.Value{}, err
+	}
+	precValue, err := result.ToInt32(precision)
+	if err != nil {
+		return result.Value{}, err
+	}
+	
+	// For general implementation
+	strValue := fmt.Sprintf("%.15f", decValue)
+	parts := strings.Split(strValue, ".")
+	
+	var resultStr string
+	
+	if precValue > 0 {
```

Comment:
I believe based on the discussion at https://github.com/cqframework/cql-tests/pull/44 and the comment linked to there, if the precision value is negative, we should be returning null.

From there we can handle the case where precisionValue == 0 separately and then handle the remaining cases.